app/models/product_import/product_importer.rb

```diff
@@ -272,5 +266,14 @@ def delete_uploaded_file
       return unless @file.path == Rails.root.join('tmp', 'product_import').to_s
       File.delete(@file)
     end
+
+    def build_entries_from_rows(rows)
+      rows.each_with_index.inject([]) do |entries, (row, i)|
+        row_data = Hash[[headers, row].transpose]
+        entry = SpreadsheetEntry.new(row_data)
+        entry.line_number = i + 2
+        entries.push entry
```

Comment:
let me ask a ruby question...
what is this entries where you call push? does it get to be @entries automagically?

Reply:
@luisramos0 It's what the block names its first argument here:

```ruby
rows.each_with_index.inject([]) do |entries, (row, i)|
  # ...
end
```

Breaking it down:

```ruby
list = rows.each_with_index # => Enumerator
list.inject([]) do |entries, row_with_index|
  row, i = row_with_index
  # ...
  entries
end
```

Reply:
ah, thanks for clarifying, I missed it.
This inject idiom is a but complex but now I understand it...